app/models/user.rb

```diff
@@ -87,7 +87,7 @@ def formatted_email
   end
 
   def password_complexity
-    return if password.blank? || password =~ /(?=.*?[#?!@$%^&*-])/
+    return if password.blank? || password =~ /(?=.*?[#?!@$%^&*-;,.()=+|:])/
```

Comment:
For this the `-` needs to be escaped. As is, the dash is saying that any character from `*` to `;` is OK, which is not the intent. That range includes numbers, which is why the third expectation is met even though I don't think it was meant to be. If you're familiar with `/a-z/` it is just like that, except `/*-;/`.

```suggestion
    return if password.blank? || password =~ /(?=.*?[#?!@$%^&*\-;,.()=+|:])/
```